### PR TITLE
Topic/swat issue 8860

### PR DESCRIPTION
**Pull request by hppritcha**
OFI and UCX fixes for issue #8860

### CODE REVIEW

**Comment by hppritcha** [comment]
@wckzhang @hoopoepg please review

**Comment by hppritcha** [comment]
@hoopoepg @wckzhang ping

**Comment by hppritcha** [comment]
@hoopoepg please review when you get a chance

**Comment by hppritcha** [comment]
@hkuno when you have time could you review?

File: opal/mca/common/ucx/common_ucx.c [hunk 1]
```diff
@@ -42,34 +47,58 @@ OPAL_DECLSPEC void opal_common_ucx_mca_var_register(const mca_base_component_t *
...
                                            MCA_BASE_VAR_SCOPE_LOCAL,
                                            &opal_common_ucx.opal_mem_hooks);
+    }
 
-        opal_common_ucx.tls = malloc(sizeof(*opal_common_ucx.tls));
+    param = mca_base_var_find("opal", "opal_common", "ucx", "tls");
+    if (0 > param) {
+
+        /*
+         * this monkey business is needed because of the way the MCA VARs framework tries to handle pointers to strings
```
Comment (hoopoepg) [inline opal/mca/common/ucx/common_ucx.c:93]: :)

**Review by hoopoepg — APPROVED** [review]

**Comment by hoopoepg** [comment]
sorry for delay - vacation :)

**Review by hkuno — APPROVED** [review]

**Pull request merged**